errors.go

```diff
@@ -67,6 +67,10 @@ var (
 
 	// ErrUnknownError is returned when notmuch returns NULL indicating an error.
 	ErrUnknownError = errors.New("unknown error occured")
+
+	// ErrNoReplies is returned if a message has no replies or if the message's C
+	// pointer did not come from a thread.
+	ErrNoReplies = errors.New("message has no replies")
```

Comment:
It's kind of awkward having this error be used for two different things. I understand it's not possible to tell the difference because of what the C library does, but I don't like the fact that the name reflects one of the possibilities but not the other.